apps/passport-server/src/services/e2eeService.ts

```diff
@@ -79,12 +85,64 @@ export class E2EEService {
       res.sendStatus(500);
     }
   }
+
+  public async handleUpdate(
+    request: UpdateE2EERequest,
+    res: Response
+  ): Promise<void> {
+    try {
+      logger(
+        `[E2EE] Updating ${request.oldBlobKey} to ${request.newBlobKey} for ${request.uuid}`
+      );
+
+      if (
+        !request.newBlobKey ||
+        !request.oldBlobKey ||
+        !request.newSalt ||
+        !request.uuid
+      ) {
+        throw new Error("Missing request fields");
+      }
+
+      // Ensure that old blob key is correct by checking if the row exists
+      const oldRow = await fetchEncryptedStorage(
+        this.context.dbPool,
+        request.oldBlobKey
+      );
+      if (!oldRow) {
+        res.sendStatus(401);
+        return;
+      }
+
+      // Ensure that new salt is different from old salt
+      const oldSalt = await this.userService.getSaltByUUID(request.uuid);
```

Comment:
i think i prefer calling something like `fetchCommitmentByUuid` rather than adding a dependency on the `userService` here.